Add a belated changelog update with recent OCaml Infrastructure updates (#2744)
ocaml/ocaml.org: data/changelog/infra/2024-10-02-updates.md

---
title: Infrastructure Updates
tags: [infrastructure]
---

> # Catching up

Reviewer: ```suggestion
# Catching Up
```
Suggested title case correction

> [Taride](https://tarides.com/)'s OCaml Infrastructure team has been hard at work

Reviewer 2: s/Taride/Tarides (I don't know if that's a typo or that how Tarides's is supposed to be written.

Reviewer: Thomas decided that any possessive of Tarides should be Tarides'

> [Taride](https://tarides.com/)'s OCaml Infrastructure team has been hard at work

Reviewer: ```suggestion
[Tarides'](https://tarides.com/) OCaml Infrastructure team has been hard at work
```

> over the last months, focused on sustaining the existing systems and
> regenerating our team's capacities. This work is in preparation for more
> substantive changes to improve the reliability and scalability of essential
> OCaml ecosystem infrastructure. We will have more fruits from this work to share
> soon, but updates on our efforts are overdue and we wanted to get you all caught
> up.

Reviewer: ```suggestion
over the last few months, focused on sustaining the existing systems and
regenerating our team's capacities. This work is in preparation for more
substantive changes to improve the reliability and scalability of essential
OCaml ecosystem infrastructure. We will have more fruit from this work to share
soon, but updates on our efforts are overdue and we wanted to get you all caught
up.
```
Minor language usage corrections

Since we last published an announcement, we have been...

> ## making the systems more stable

Reviewer: ```suggestion
## Making the Systems More Stable
```

- We have maintained opam repo CI at an average of 99.9% availability for the
  last 2 months.

- The Ansible scripts that configure our services has been put under continuous
  deployment via [@mtelvers](https://github.com/mtelvers) work on the
  https://github.com/ocurrent/ocurrent-configurator utility.

- We've added a linting step to ensure packages include an email address for
  maintainers. This is needed to ensure opam repo curators can reliably contact
  package maintainers when needed (by
  [@punchagan](https://github.com/punchagan), see
  [ocurrent/opam-ci-check#30](https://github.com/ocurrent/opam-ci-check/pull/30)).

- There was a profusion of solver timeout errors impacting the opam repo CI
  system, and we have managed to mitigate these (by
  [@mtelvers](https://github.com/mtelvers) and

>   [@shonfeder](https://github.com/shonfeder) see

Reviewer: ```suggestion
  [@shonfeder](https://github.com/shonfeder), see
```

  [ocaml/infrastructure#147](https://github.com/ocaml/infrastructure/issues/147).

- We fixed an uncaught exception that sometimes occurred during linting checks
  in the opam repo CI (by [@shonfeder](https://github.com/shonfeder), see
  [ocurrent/opam-repo-ci#341](https://github.com/ocurrent/opam-repo-ci/pull/341).

> ## keeping the systems up to date

Reviewer: ```suggestion
## Keeping the Systems Up to Date
```

- Updates to address [CVE-2024-6387](https://www.cve.main
  /CVERecord?id=CVE-2024-6387) (by [@mtelvers](https://github.com/mtelvers), see
  [ocaml/infrastructure#140](https://github.com/ocaml/infrastructure/issues/140)).

- Updates to support OCaml 5.2 and 5.3 (by
  [@mtelvers](https://github.com/mtelvers), see
  [ocurrent/opam-repo-ci#268](https://github.com/ocurrent/opam-repo-ci/issues/268)
  and
  [ocurrent/opam-repo-ci#363](https://github.com/ocurrent/opam-repo-ci/issues/363)).

- Updates for better support for Windows (by
  [@mtelvers](https://github.com/mtelvers), see
  [ocurrent/docker-base-images#241](https://github.com/ocurrent/docker-base-images/issues/241)).

- Support for GCC 14, culminating in a patch for the compiler (by
  [@mtelvers](https://github.com/mtelvers), see
  [ocurrent/docker-base-images#279](https://github.com/ocurrent/docker-base-images/issues/279)).

- Dropped support for the post-LTS distro versions Debian 10 and 11 (by
  [@mtelvers](https://github.com/mtelvers) and

>   [@shonfeder](https://github.com/shonfeder) see

Reviewer: ```suggestion
  [@shonfeder](https://github.com/shonfeder), see
```

  [ocurrent/ocaml-dockerfile#220](https://github.com/ocurrent/ocaml-dockerfile/pull/220)
  and
  [ocurrent/ocaml-dockerfile#210](https://github.com/ocurrent/ocaml-dockerfile/pull/210)).

> ## making it easier to understand the systems

Reviewer: ```suggestion
## Making It Easier to Understand the Systems
```

- We are [generating documentation of the platforms under test in opam
  CI][platforms] (by [@benmandrew](https://github.com/benmandrew)).

- We are [generating documentation of the services under continuous
  deployment][services] (by [@benmandrew](https://github.com/benmandrew)).

[services]: https://github.com/ocurrent/ocurrent-deployer/blob/master/doc/services.md
[platforms]: https://github.com/ocurrent/opam-repo-ci/blob/master/doc/platforms.md

> ## improving the functionality and usability of the systems

Reviewer: ```suggestion
## Improving the Functionality and Usability of the Systems
```

- We have been developing

>   [opam-ci-check](https://github.com/ocurrent/opam-repo-ci/tree/master/opam-ci-check),

Reviewer: ```suggestion
  [`opam-ci-check`](https://github.com/ocurrent/opam-repo-ci/tree/master/opam-ci-check),
```

  a command line utility aiming to make opam repo CI's logic local-first. It is
  still in the early stages and far from finished, but it is enabling faster

>   development iteration already, and it will soon be fit for early adopting

Reviewer 2: I think we can already call it fit for early adopters to test, given that we have added instructions to reproduce locally. (to be deployed soon)?

  package authors to use locally or in their own CI pipelines (by
  [@punchagan](https://github.com/punchagan) and
  [@shonfeder](https://github.com/shonfeder)).

- The first benefits of `opam-ci-check`'s local-first approach have made their
  way into opam CI as logged instructions for running the lint checks locally
  and as improved error reporting (by
  [@punchagan](https://github.com/punchagan), see
  [ocurrent/opam-repo-ci#359](https://github.com/ocurrent/opam-repo-ci/issues/359)
  and
  [ocurrent/opam-repo-ci#360](https://github.com/ocurrent/opam-repo-ci/issues/360)).

> # Forging ahead

Reviewer: ```suggestion
# Forging Ahead
```
